make_site.go

```diff
@@ -67,16 +75,16 @@ func makeSiteStruct(objs []Object) {
 	}
 }
 
-func makeSitemap(objs []Object) {
+func makeSitemap(objs map[string]*Object) {
 	t := template.Must(template.ParseFiles("tmpl/sitemap-tmpl.xml"))
 	f, _ := os.Create("tmpl/sitemap.xml")
 	t.Execute(f, objs)
 }
 
-func makeObjById(selector string, s *goquery.Selection) (obj Object) {
+func makeObjById(selector string, s *goquery.Selection) (obj *Object) {
 	s.Find(selector).Each(func(_ int, s *goquery.Selection) {
 		desc := s.NextFiltered("p")
-		ul := desc.NextFiltered("ul")
+		ul := s.NextFilteredUntil("ul", "h2")
```

Comment:
This avoids empty subpages, as but several linked pages on awesome-go.com are just empty when they are in a nested list, i.e. Middlewares -> Actual Middlewares.